app/controllers/orders_controller.rb

```diff
@@ -1,15 +1,47 @@
 class OrdersController < ApplicationController
   before_action :require_logged_in, only: [:new]
-  before_action :load_info_cart, only: [:new]
+  before_action :load_products_in_cart, only: [:new]
+  before_action :load_total, only: [:new]
 
   def new; end
 
+  def create
+    ActiveRecord::Base.transaction do
+      order = current_user.orders.create(order_params)
+      order_details = order.order_details.build load_order_details
+      save_order_details order_details
+      session[:cart] = {}
+
+      redirect_to root_path, notice: t(".create_order_success")
+    rescue StandardError
+      flash[:error] = t ".create_order_error"
+      raise ActiveRecord::Rollback
```

Comment:
railse exception để chi thế cậu, xóa thử dc ko, và ngoài ra, sau khi add flash thì nên di chuyển đi đâu nhé

Reply:
Rails exception để handle việc phục hồi dữ liệu trở về ban đầu nếu việc thay đổi dữ liệu database có lỗi, đảm bảo toàn vẹn dữ liệu.

Reply:
dòng này ko cần em ah, vì em đang dùng các method raise exception rồi